python/semantic_kernel/connectors/ai/embeddings/embedding_generator_base.py

```diff
@@ -3,6 +3,8 @@
 from abc import ABC, abstractmethod
 from typing import TYPE_CHECKING, Any
 
+from typing_extensions import deprecated
```

Comment:
ideally we already do the sys.version check for python 3.13 here: https://docs.python.org/3/library/warnings.html#warnings.deprecated